### PR TITLE
test: add `Test.Handler.Gen.X.flix`

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes https://github.com/flix/flix/issues/7060

### CODE REVIEW

**Comment by jaschdoc** [comment]
Should we drop `AtomicInteger` or is it okay? I think it's awkward with all the conversion going on between `AtomicInteger` and `Int32`.

**Comment by jaschdoc** [comment]
@magnus-madsen I think I have everything covered now.

- `Unit`
- `Bool`
- `Char`
- `String`
- `Int8-64`
- `Float32-64`
- `Option`
- `AtomicInteger`

File: main/test/flix/Test.Handler.Gen.AtomicInteger.flix [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        intValue(n)
+
+    eff Gen {
+        pub def gen(): AtomicInteger
+    }
+
+    def generator(): Unit \ Gen =
+        do Gen.gen(); generator()
+
+    def sample(limit: Int32): List[Int32] =
```
Comment (magnus-madsen) [inline main/test/flix/Test.Handler.Gen.AtomicInteger.flix:20]: I would be better for the test case if the return type was List[AtomicInteger].

File: main/test/flix/Test.Handler.Gen.Int8.flix [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                    if (i == limit) Nil else i :: k(i)
+            }
+        }
+
+    def getAndInc(r: Ref[Int8, r]): Int8 \ r =
+        let i = deref r;
+        r := (i + 1i8);
+        i
+
+    def range(a: Int8, b: Int8): List[Int8] =
```
Comment (magnus-madsen) [inline main/test/flix/Test.Handler.Gen.Int8.flix:27]: I guess the problem is that List.range is only for Int32s.

Comment (magnus-madsen) [reply main/test/flix/Test.Handler.Gen.Int8.flix:27]: So you need your own.

File: main/test/flix/Test.Handler.Gen.Char.flix [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            try {
+                generator(); Nil
+            } with Gen {
+                def gen(k) =
+                    let i = getAndInc(counter);
+                    if (i == limit) Nil else i :: k(i)
+            }
+        }
+
+    def next(c: Char): Char = match c {
```
Comment (magnus-madsen) [inline main/test/flix/Test.Handler.Gen.Char.flix:22]: Seems reasonable.

**Review by magnus-madsen — APPROVED** [review]

File: main/test/flix/Test.Handler.Gen.AtomicInteger.flix (hunk 1 above)
Comment (magnus-madsen) [reply main/test/flix/Test.Handler.Gen.AtomicInteger.flix:20]: Lets fix this in a subsequent PR.

**Pull request merged**